penguin/integrations/mcp/server_tools/blueprints.py

```
@@ -0,0 +1,370 @@
+
+    blueprint_path = _optional_str(arguments.get("blueprint_path"))
+    if blueprint_path is None:
+        raise ValueError("blueprint_path or content is required")
+    path = Path(blueprint_path).expanduser().resolve()
+    if not path.exists():
+        raise FileNotFoundError(f"Blueprint file not found: {path}")
+    parser = BlueprintParser(base_path=path.parent)
+    return parser.parse_file(path), str(path)
+
+
+def _parse_content(
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_parse_content`

(ANN202)